drivers/soundwire/intel.c

```diff
@@ -867,6 +873,33 @@ static int intel_prepare(struct snd_pcm_substream *substream,
 		return -EIO;
 	}
 
+	if (dma->suspend_reinitialize) {
+		/*
+		 * .prepare() is called when dealing with underflows or
+		 * after system resume. In the latter case, we need to
+		 * re-initialize the ALH settings, which is not needed
+		 * in the underflow case
+		 */
+
+		/* configure stream */
+		ch = params_channels(sdw->hw_params);
+		if (substream->stream == SNDRV_PCM_STREAM_CAPTURE)
+			dir = SDW_DATA_DIR_RX;
+		else
+			dir = SDW_DATA_DIR_TX;
+
+		intel_pdi_shim_configure(sdw, sdw->pdi);
+		intel_pdi_alh_configure(sdw, sdw->pdi);
+		sdw_cdns_config_stream(cdns, ch, dir, sdw->pdi);
+
+		/* Inform DSP about PDI stream number */
+		ret = intel_params_stream(sdw, substream, dai, sdw->hw_params,
+					  sdw->instance,
+					  sdw->pdi->intel_alh_id);
+
+		dma->suspend_reinitialize = false;
```

Comment:
@plbossart dont you also have to set this to false in intel_hw_params()?

Reply:
done